src/core/compact_object.cc

```diff
@@ -1255,4 +1232,31 @@ MemoryResource* CompactObj::memory_resource() {
   return tl.local_mr;
 }
 
+constexpr std::pair<CompactObjType, std::string_view> kObjTypeToString[8] = {
+    {OBJ_STRING, "string"sv},  {OBJ_LIST, "list"sv},    {OBJ_SET, "set"sv},
+    {OBJ_ZSET, "zset"sv},      {OBJ_HASH, "hash"sv},    {OBJ_STREAM, "stream"sv},
+    {OBJ_JSON, "ReJSON-RL"sv}, {OBJ_SBF, "MBbloom--"sv}};
+
+std::optional<std::string_view> ObjTypeToString(CompactObjType type) {
+  for (auto& p : kObjTypeToString) {
+    if (type == p.first) {
+      return p.second;
+    }
+  }
+
+  LOG(ERROR) << "Unsupported type " << type;
+  return std::nullopt;
+}
+
+std::optional<CompactObjType> ObjTypeFromString(std::string_view sv) {
+  for (auto& p : kObjTypeToString) {
+    if (absl::EqualsIgnoreCase(sv, p.second)) {
+      return p.first;
+    }
+  }
+
+  LOG(ERROR) << "Unsupported type " << sv;
```

Comment:
similarly you do not log ERROR in low level utility functions that lack context. 
Especially when your API is designed to handle the "unsupported" case by returning nullopt. 

It's the responsibility of the caller that knows the context to either log or reply the error or whatever. 

If it is EXPECTED that `sv` is a valid string type then see my comment above otherwise you can keep the optional in this case.

Reply:
No, everything is okay, I got it.